lte/gateway/c/oai/tasks/nas/nas_state_converter.cpp

```diff
@@ -53,19 +53,106 @@ void NasStateConverter::partial_tai_list_to_proto(
   partial_tai_list_proto->set_type_of_list(state_partial_tai_list->typeoflist);
   partial_tai_list_proto->set_number_of_elements(
       state_partial_tai_list->numberofelements);
-  // TODO
+  switch (state_partial_tai_list->typeoflist) {
+    case TRACKING_AREA_IDENTITY_LIST_MANY_PLMNS: {
+      for (int idx = 0; idx < TRACKING_AREA_IDENTITY_LIST_MAXIMUM_NUM_TAI;
+           idx++) {
+        oai::Tai* proto_many_plmn = partial_tai_list_proto->add_tai_many_plmn();
+        tai_to_proto(
+            &state_partial_tai_list->u.tai_many_plmn[idx], proto_many_plmn);
+      }
+    } break;
+    case TRACKING_AREA_IDENTITY_LIST_ONE_PLMN_CONSECUTIVE_TACS: {
+      tai_to_proto(
+          &state_partial_tai_list->u.tai_one_plmn_consecutive_tacs,
+          partial_tai_list_proto->mutable_tai_one_plmn_consecutive_tacs());
+    } break;
+    case TRACKING_AREA_IDENTITY_LIST_ONE_PLMN_NON_CONSECUTIVE_TACS: {
+      char plmn_array[PLMN_BYTES];
+      plmn_array[0] =
+          (char) (state_partial_tai_list->u.tai_one_plmn_non_consecutive_tacs.mcc_digit1 + ASCII_ZERO);
+      plmn_array[1] =
+          (char) (state_partial_tai_list->u.tai_one_plmn_non_consecutive_tacs.mcc_digit2 + ASCII_ZERO);
+      plmn_array[2] =
+          (char) (state_partial_tai_list->u.tai_one_plmn_non_consecutive_tacs.mcc_digit3 + ASCII_ZERO);
+      plmn_array[3] =
+          (char) (state_partial_tai_list->u.tai_one_plmn_non_consecutive_tacs.mnc_digit1 + ASCII_ZERO);
+      plmn_array[4] =
+          (char) (state_partial_tai_list->u.tai_one_plmn_non_consecutive_tacs.mnc_digit2 + ASCII_ZERO);
+      plmn_array[5] =
+          (char) (state_partial_tai_list->u.tai_one_plmn_non_consecutive_tacs.mnc_digit3 + ASCII_ZERO);
+      partial_tai_list_proto->set_plmn(plmn_array);
+      for (int idx = 0; idx < TRACKING_AREA_IDENTITY_LIST_MAXIMUM_NUM_TAI;
+           idx++) {
+        partial_tai_list_proto->add_tac(
+            state_partial_tai_list->u.tai_one_plmn_non_consecutive_tacs
+                .tac[idx]);
+      }
+    } break;
+  }
 }
 
 void NasStateConverter::tai_list_to_proto(
     const tai_list_t* state_tai_list, oai::TaiList* tai_list_proto) {
   tai_list_proto->set_numberoflists(state_tai_list->numberoflists);
-  // TODO
+  for (int idx = 0; idx < state_tai_list->numberoflists; idx++) {
+    oai::PartialTaiList* partial_tai_list =
+        tai_list_proto->add_partial_tai_lists();
+    partial_tai_list_to_proto(
+        &state_tai_list->partial_tai_list[idx], partial_tai_list);
+  }
+}
+
+void NasStateConverter::proto_to_partial_tai_list(
```

Comment:
To make the code more readable and easy to debug, please put the deserialization function definition right after the serialization function.